### PR TITLE
[CALCITE-3527] Enrich tests for sql hints.

### PR DESCRIPTION
**Pull request by cshuo**
* fix problems in `SqlHintsConverterTest`
* add more tests to cover functionality  of sql hint.

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/test/SqlHintsConverterTest.java [hunk 1]
```diff
@@ -112,12 +113,14 @@ protected DiffRepository getDiffRepos() {
     sql(sql).ok();
   }
 
-  @Test public void testInvalidQueryHint() {
+  @Test public void testInvalidQueryHint1() {
     final String sql = "select /*+ weird_hint */ empno\n"
```
Comment (danny0405) [inline core/src/test/java/org/apache/calcite/test/SqlHintsConverterTest.java]: Thanks for the refactoring, can we give the test cases more meaningful names ?

File: core/src/test/java/org/apache/calcite/test/SqlHintsConverterTest.java [hunk 2]
```diff
@@ -185,20 +197,46 @@ protected DiffRepository getDiffRepos() {
...
+  @Test public void testInvalidTableHints1() {
     final String sql = "select\n"
         + "ename, job, sal, dept.name\n"
         + "from emp /*+ weird_hint(idx1, idx2) */\n"
         + "join dept /*+ properties(k1='v1', k2='v2') */\n"
         + "on emp.deptno = dept.deptno";
     sql(sql).fails("Hint: WEIRD_HINT should be registered in the HintStrategies.");
+  }
 
+  @Test public void testInvalidTableHints2() {
```
Comment (danny0405) [inline core/src/test/java/org/apache/calcite/test/SqlHintsConverterTest.java]: Can we also add test cases with query that have 3, 4 or more nested query levels ? Also you can add some test cases for decorrelation.

File: core/src/test/java/org/apache/calcite/test/SqlHintsConverterTest.java (hunk 1 above)
Comment (cshuo) [reply core/src/test/java/org/apache/calcite/test/SqlHintsConverterTest.java]: make sense.

File: core/src/test/java/org/apache/calcite/test/SqlHintsConverterTest.java (hunk 2 above)
Comment (cshuo) [reply core/src/test/java/org/apache/calcite/test/SqlHintsConverterTest.java]: ok.

**Comment by julianhyde** [comment]
Regarding the commit message: SQL is upper-case, and don't end the sentence with a '.'.

**Comment by cshuo** [comment]
> Regarding the commit message: SQL is upper-case, and don't end the sentence with a '.'.

ok, got it.

**Comment by cshuo** [comment]
checkstyle failed...on updating.

**Pull request closed without merging**